source/hooks/useAsync.ts

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from "react";
+import { to } from "../helpers/Misc";
+
+export default function useAsync<T>(
+  asyncFunction: () => Promise<T>
+): [boolean, T | undefined, Error | undefined] {
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
Perhaps the asyncFunction should be a state instead. By doing this we avoid that the asyncFunction is re-triggered when the function is changed outside this hook.